stonesoup/types/hypothesis.py

```diff
@@ -101,6 +101,10 @@ def weight(self):
 class SingleProbabilityHypothesis(ProbabilityHypothesis, SingleHypothesis):
     """Single Measurement Probability scored hypothesis subclass."""
 
+    def __hash__(self):
+        return hash((self.probability, self.prediction, self.measurement,
+                    self.measurement_prediction))
+
```

Comment:
Is this required? What about other hypothesies types if so.

Reply:
Possibly not. As far as I can see the hypotheses are never keys in a dictionary, so this step should hopefully be reversible. I shall comment it out and see if the tests will run.